operators/src/plot/statistics.rs

```diff
@@ -161,20 +161,15 @@ impl PlotQueryProcessor for StatisticsQueryProcessor {
 }
 
 #[allow(clippy::float_cmp)] // allow since NO DATA is a specific value
-fn process_raster(number_statistics: &mut NumberStatistics, tile_grid: &Grid2D<f64>) {
-    let no_data_value = tile_grid.no_data_value();
-
-    if let Some(no_data_value) = no_data_value {
-        for &value in &tile_grid.data {
-            if value == no_data_value {
-                number_statistics.add_no_data();
-            } else {
-                number_statistics.add(value);
-            }
-        }
-    } else {
-        for &value in &tile_grid.data {
+fn process_raster<I>(number_statistics: &mut NumberStatistics, data: I)
+where
+    I: Iterator<Item = Option<f64>>,
+{
+    for value_option in data {
+        if let Some(value) = value_option {
```

Comment:
question: for rasters with no no data value we didn't have to check whether they are no data. Now we have to check it always. Would it make sense to have a grid type where we know there is data everywhere (like in the empty case where we know there is only nodata) and implement operations more efficiently. Here for example I guess even on an empty grid we would iterate over a bunch of None pixels. Can we avoid that?

edit: nevermind, emptygrids are handlet completely difference ind line 149 I see now. Still the question with tiles with no nodata remains.

Reply:
good question. We can add the Grid(Grid) variant to the GridOrEmpty enum. The first time a Grid(Grid) is used in an operator where empty pixels can be a result it has to be converted into a MAskedGrid in order to be able to use parallel processing. Then we can check if all validity flags are true and turn it back. i think it is not worth it.

Reply:
so, should we add this as a card or consider it a moonshot and get to it whenever?